### PR TITLE
Support for new tagging changes

### PR DESCRIPTION
**Pull request by mkanoor**
Uses the latest tagging changes 
* tag end point to create and assign new tags
* untag end point to delete tags

Waiting on some of the other PRs to get merged first

https://github.com/RedHatInsights/catalog-api/pull/495

### CODE REVIEW

File: app/services/workflow_find_service.rb [hunk 1]
```diff
@@ -26,8 +26,6 @@ def find_by_tag_resources(tag_resources)
   end
 
   def fq_tag_names(tag_attrs)
-    GetRemoteTags.new(tag_attrs).process.tags.collect do |tag|
-      "/#{tag[:namespace]}/#{tag[:name]}=#{tag[:value]}"
-    end
+    GetRemoteTags.new(tag_attrs).process.tags.collect { |tag| tag[:tag] }
```
Comment (hsong-rh) [inline app/services/workflow_find_service.rb]: You sure you need an another level of calling `tag` attribute? It's already called in `GetRemoteTags.new(tag_attrs).process.tags`

File: app/services/get_remote_tags.rb [hunk 2]
```diff
@@ -7,14 +7,7 @@ def initialize(options)
 
   def process
     response = get_request(object_url)
-    build_tags(response.body)
+    @tags = JSON.parse(response.body)['data'].collect { |tag| { :tag => tag['tag'] } }
```
Comment (bzwei) [inline app/services/get_remote_tags.rb]: What is the data type of `JSON.parse(response.body)['data']`? Is it already an array? Do you simply convert the key from `'tag'` to `:tag`?

File: app/services/workflow_find_service.rb [hunk 3]
```diff
@@ -7,13 +7,13 @@ def find(tag_attrs)
...
 
-  # find workflows from a collection of [app_name, object_type, [namespace, key, value]]
+  # find workflows from a collection of [app_name, object_type, [tags]]
   def find_by_tag_resources(tag_resources)
     return [] if tag_resources.blank?
 
     query = nil
     tag_resources.each_with_index do |tr, i|
-      tag_names = tr['tags'].map { |tag| "/#{tag['namespace']}/#{tag['name']}=#{tag['value']}" }
+      tag_names = tr['tags'].collect { |tag| tag[:tag] }
```
Comment (bzwei) [inline app/services/workflow_find_service.rb]: not sure why we have `tag[:tag]`. shouldn't be `tag['tag']`?
Don't understand why we need to have format of `[{"tag": "tag1"}, {"tag": "tag2"}]`, which can be simply `["tag1", "tag2"]`. I realize this format comes from external services, but still don't understand why.

File: spec/controllers/v1.0/requests_controller_spec.rb [hunk 4]
```diff
@@ -459,7 +459,7 @@
       [{
         'app_name'    => 'app1',
         'object_type' => 'otype1',
-        'tags'        => [{'namespace' => 'ns1', 'name' => 'name1', 'value' => 'v1'}]
+        'tags'        => [{:tag => '/ns1/name1=v1'}]
```
Comment (bzwei) [inline spec/controllers/v1.0/requests_controller_spec.rb]: prefer `'tag'` other than `:tag`

Comment (bzwei) [reply spec/controllers/v1.0/requests_controller_spec.rb]: I would like to define request body to have `"tags": ["/ns1/name1=v1", "/ns2/name2=v2"]`
I understand the app that prepares the request body get tags from other API with the format `[{"tag": "/ns1/name1=v1"}, {"tag":"/ns2/name2=v2"}]`. Approval-api has to accept the redundant format if it does not want the caller to have the burden to do a format transformation.

Comment (mkanoor) [reply spec/controllers/v1.0/requests_controller_spec.rb]: @bzwei @hsong-rh 
The multiple tag is  from the Tagging changes implemented here
https://github.com/RedHatInsights/topological_inventory-api/blob/784bc2640efd190d3c43b3ce5ebaa16f6125906a/public/doc/openapi-3-v2.0.0.json#L10696

File: app/services/add_remote_tags.rb [hunk 5]
```diff
@@ -1,6 +1,10 @@
 class AddRemoteTags < RemoteTaggingService
   def process(tag)
-    post_request(object_url, tag)
+    post_request(object_url, [tag])
```
Comment (bzwei) [inline app/services/add_remote_tags.rb]: The class name `AddRemoteTags`, and the change here indicates a single tag. Should we change the class name or allow input to be an array?

File: app/services/delete_remote_tags.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class DeleteRemoteTags < RemoteTaggingService
```
Comment (bzwei) [inline app/services/delete_remote_tags.rb]: Same question about the class name vs input argument.

[the remark this answers is not in the document]
Comment (mkanoor) [reply app/services/delete_remote_tags.rb]: @hsong-rh They are different its using inheritance to share the common code. The object_url keeps changing based on what your action is

File: app/services/remote_tagging_service.rb [hunk 7]
```diff
@@ -36,7 +41,7 @@ def self.sources_url
   private_class_method :sources_url
 
   def object_url
-    "#{service_url}/#{@object_type.underscore.pluralize}/#{@object_id}/tags"
+    "#{service_url}/#{@object_type.underscore.pluralize}/#{@object_id}/tags?limit=#{QUERY_LIMIT}&#{approval_tag_filter}"
```
[the remark this answers is not in the document]
Comment (mkanoor) [reply app/services/remote_tagging_service.rb]: @bzwei @hsong-rh The object_url keeps changing based on the action when we want to add_remote_tags the end point ends in "tag" when its being deleted it ends in "untag"

**Comment by mkanoor** [comment]
@bzwei @hsong-rh 
Please review

File: app/services/add_remote_tags.rb (hunk 5 above)
Comment (mkanoor) [reply app/services/add_remote_tags.rb]: @bzwei we are adding one tag at a time when we do the link and also when we do the unlink. So its still singular for us might be plural for remote services

File: app/services/delete_remote_tags.rb (hunk 6 above)
Comment (mkanoor) [reply app/services/delete_remote_tags.rb]: @bzwei We are unlink one workflow at a time, when we unlink multiples we can can change the name

Comment (mkanoor) [reply app/services/delete_remote_tags.rb]: @bzwei Missed it I will fix the inputs

**Pull request merged**